Automattic/simplenote-android
# Note List Widget

### Fix
Add both dark and light note list widgets to close https://github.com/Automattic/simplenote-android/issues/232.  The widgets show all notes in the order they are sorted within the app.  The widgets show a ***New Note*** button in the lower, right/end of the widget if the height is larger than two list items and the width is larger than double the height.  When the widget is resized, the button will be shown or hidden based on the dimensions.  See the screenshots and animation below for illustration.

![widget_note_list](https://user-images.githubusercontent.com/3827611/74679358-df27aa00-517a-11ea-80fc-6ca0097242ac.png)

<a href="https://user-images.githubusercontent.com/3827611/74679421-07170d80-517b-11ea-988a-2e917f178b2e.gif"><img src="https://user-images.githubusercontent.com/3827611/74679421-07170d80-517b-11ea-988a-2e917f178b2e.gif" width="300"></a>

#### Note
These changes have a lot of duplicated code since this was a small, time-boxed project. Ideally, the duplication would be consolidated, but we want to see more usage of the note widgets before investing time into optimizing the code.

### Test
These steps assume the AOSP launcher (i.e. Nexus/Pixel launcher). The steps to add the widget to the home screen may be slightly different based on the launcher used. Other steps should be the same regardless of launcher.
1. Long-press home screen.
2. Tap ***Widgets*** option.
3. Scroll to ***Simplenote*** widget.
4. Notice ***Note List (Dark)*** and ***Note List (Light)*** widgets.
5. Long-press ***Note List (Dark)*** widget.
6. Place ***Note List (Dark)*** widget on home screen.
7. Notice note list shown in widget with dark theme.
8. Long-press widget.
9. Notice circular handles are shown.
10. Drag circular handles horizontally and vertically.
11. Notice note button is shown in widget with dark theme when large enough.
12. Notice note list is vertically scrollable.
13. Tap any note in widget.
14. Notice note is shown in app.
15. Go to home screen.
16. Tap button in widget.
17. Notice new note is shown in app.

### Review
Only one developer and one designer are required to review these changes, but anyone can perform the review.

### Release
`RELEASE-NOTES.txt` was updated in 9855b7c with:
> Added a note list widget to open any note in the app from the home screen
